store/localcachelayer/channel_layer.go

```diff
@@ -113,6 +130,63 @@ func (s LocalCacheChannelStore) GetMemberCountFromCache(channelId string) int64
 	return count
 }
 
+// ChannelCacheByName methods
+func (s LocalCacheChannelStore) GetByName(teamId string, name string, allowFromCache bool) (*model.Channel, *model.AppError) {
+	return s.getByName(teamId, name, false, allowFromCache)
+}
+
+func (s LocalCacheChannelStore) GetByNames(teamId string, names []string, allowFromCache bool) ([]*model.Channel, *model.AppError) {
+	var channels []*model.Channel
+
+	if allowFromCache {
+		var misses []string
+		visited := make(map[string]struct{})
+		for _, name := range names {
+			if _, ok := visited[name]; ok {
+				continue
+			}
+			visited[name] = struct{}{}
+			if cacheItem := s.rootStore.doStandardReadCache(s.rootStore.channelByNameCache, teamId+name); cacheItem != nil {
```

Comment:
We have changed how we count the hits and misses in this part of the code, to have a better separation between `GetByName` and `GetByNames`, so that should be reflected here. I would expect to have the changes incorporated during the merge.